src/background.ts

```diff
@@ -152,13 +161,41 @@ function fetchEngineInfosFromUserDirectory(): EngineInfo[] {
       path: engineDir,
       executionEnabled: true,
       executionFilePath: path.join(engineDir, manifest.command),
+      type: "userDir",
+    });
+  }
+  for (const engineDir of store.get("enginePaths")) {
+    if (!fs.existsSync(engineDir)) {
+      dialog.showErrorBox(
+        "エンジンが見つかりませんでした",
+        `${engineDir}が見つかりませんでした。このエンジンの登録は解除されます。`
+      );
+      store.set(
+        "enginePaths",
+        store.get("enginePaths").filter((p) => p !== engineDir)
+      );
+      continue;
```

Comment:
見つからなかった場合、破壊的に消去するのではなくスルーしてwarningだけ表示するのはどうでしょう。
（消えてほしくない人もいそうなので）

Reply:
スルーするとEngineInfoに入れられず、そのためUIに出なくなって消せなくなるんですよね。
だから消しちゃった方が良いかなと思いました。

Reply:
あー、なるほどです！
たぶん理想の動作としては「ここで消すかどうかも選べて、後でエンジン管理からも消せる」だと思うのですが、実装がそこそこ煩雑になっちゃいますね･･･。

なぜ消去するかここにTODOコメントを書いとくのはどうでしょう。
あと他の箇所では「エンジンを削除」と呼んでいたので統一してみました。（もしかしたら「解除」で統一したほうが良いかも）
```suggestion
      // 動かないエンジンは追加できないので削除
      // FIXME: エンジン管理UIで削除可能にする
      dialog.showErrorBox(
        "エンジンが見つかりませんでした",
        `${engineDir}が見つかりませんでした。このエンジンは削除されます。`
      );
      store.set(
        "enginePaths",
        store.get("enginePaths").filter((p) => p !== engineDir)
      );
      continue;
```